### ISSUE TITLE
Napari fails to find dock widgets for plugins with dots in the name

### ISSUE THREAD

**Issue by kmdouglass** [napari/npe2#232]
* npe2 version: 0.6.1
* napari version: 0.4.16
* Python version: 3.10.4
* Operating System: Ubuntu 20.04

### Description

PRs #227 and #230 added the ability to create plugins with dots in the name. With these changes `npe2 validate` successfully validates our Napari yaml manifest file.

Unfortunately, Napari fails to load these plugins, presumably due to an error in plugin discovery.

CC @elio01epfl

### What I Did

After installing our plugin with `pip install -e .` from the root of our plugin's code repository, we launched Napari with the following script:

```python
import napari
import os

viewer = napari.Viewer() 
path=  str(os.path.dirname(__file__))+'/images/steven_192.ome.zarr/Images'#"https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.3/9836842.zarr/"
viewer.open(path, plugin = 'napari-ome-zarr')
viewer.window.add_plugin_dock_widget('leb.eda-napari','Plot frame rate') #'Add time scroller'
napari.run()
```

We see the following on stdout/stderr:

```console
Warnings:
Invalid schema for package 'leb.eda-napari', please run 'npe2 validate leb.eda-napari' to check for manifest errors.
Invalid schema for package 'leb.eda-napari', please run 'npe2 validate leb.eda-napari' to check for manifest errors.
Traceback (most recent call last):
  File "/home/elio/Desktop/LEB_Internship/napariLEB/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/elio/Desktop/LEB_Internship/napariLEB/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/elio/.vscode/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/home/elio/.vscode/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/elio/.vscode/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File "/home/elio/.vscode/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 321, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/elio/.vscode/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 135, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/elio/.vscode/extensions/ms-python.python-2022.12.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "/home/elio/Desktop/LEB_Internship/eda-napari/debug.py", line 11, in <module>
    viewer.window.add_plugin_dock_widget('leb.eda-napari','Plot frame rate') #'Add time scroller'
  File "/home/elio/Desktop/LEB_Internship/napariLEB/lib/python3.10/site-packages/napari/_qt/qt_main_window.py", line 683, in add_plugin_dock_widget
    Widget, dock_kwargs = plugin_manager.get_widget(
  File "/home/elio/Desktop/LEB_Internship/napariLEB/lib/python3.10/site-packages/napari/plugins/_plugin_manager.py", line 617, in get_widget
    raise KeyError(msg)
KeyError: "Plugin 'leb.eda-napari' does not provide any dock widgets"
```

We poked around the code with the debugger, and it looks like the `_dock_widgets` attribute of the Napari plugin manager isn't getting populated with our plugin's info: https://github.com/napari/napari/blob/main/napari/plugins/_plugin_manager.py#L603

It's also interesting to note the warnings about the invalid schema for the plugin because it passes `npe2 validate` without any issues.

At this point I'm not certain whether the problem is with our code, `npe2`, or `napari` itself. Any ideas?

Thanks a lot!

#### napari.yaml

```
name: leb.eda-napari
contributions:
  commands:
    - id: leb.eda-napari.frame_rate_widget
      title: open my widget
      python_name: leb.eda_napari._widget:Frame_rate_Widget
    - id: leb.eda-napari.Time_scroller_widget
      title: add time scroller widget
      python_name: leb.eda_napari._widget:Time_scroller_widget
  widgets:
    - command: leb.eda-napari.frame_rate_widget 
      display_name: Plot frame rate
    - command: leb.eda-napari.Time_scroller_widget
      display_name: Add time scroller
```

**Comment by tlambert03** [comment]
Thanks @kmdouglass,

I suspect the assumption of a single period delimiter made it into the code here somewhere too.  Will have a look

**Comment by tlambert03** [comment]
> We poked around the code with the debugger, and it looks like the _dock_widgets attribute of the Napari plugin manager isn't getting populated with our plugin's info: https://github.com/napari/napari/blob/main/napari/plugins/_plugin_manager.py#L603

That's actually the old "first generation" plugin manager.  While we transition, we need to maintain support for "npe1" plugins. The relevant plugin manager here is actually the one in the npe2 repo, and I'm confident that one would "know" about the dock widget.  

On the napari side, it's plugins/_npe2.py that we should look at to see why iter_widgets wouldn't match (and it's probably the plugin name being stripped to the part before the period)

**Comment by kmdouglass** [comment]
> That's actually the old "first generation" plugin manager. While we transition, we need to maintain support for "npe1" plugins. The relevant plugin manager here is actually the one in the npe2 repo, and I'm confident that one would "know" about the dock widget.

Ok I see. It was the traceback that pointed us to look at `napari/plugins/_plugin_manager.py`, and we confirmed that we were inside some of those methods during the debugger session. Maybe we're entering the wrong plugin manager? Do you think we should try working off the head of Napari's main branch instead of the 0.4.16 release?

Either way, I defer to your expertise here. Thanks again 😁

**Comment by tlambert03** [comment]
this was cause by this line: https://github.com/napari/npe2/blob/a2ab7494613b8e7c00c1b54abdebc364a5325ebf/npe2/manifest/utils.py#L80 ... which would have returned a faulty `plugin_name` for dotted names, causing napari to skip over it in this line: https://github.com/napari/napari/blob/a23740339b1e78822f6c6996f05973d3f5bef2df/napari/plugins/_npe2.py#L114

fixed in #233

**Comment by nclack** [comment]
fixed in #238 (which came out of #233)

**Comment by kmdouglass** [comment]
Hi @tlambert03 ,
We tested packaging a plugin with a dot in the package name from the current head of `main` 4b7f0fe and it works like a charm.

Thanks again!
